src/gui/Src/BasicView/StdIconSearchListView.cpp

```diff
@@ -18,6 +18,6 @@ void StdIconSearchListView::setIconColumn(int c)
 
 void StdIconSearchListView::setRowIcon(int r, const QIcon & icon)
 {
-    clearFilter();
+    //clearFilter();

```

Comment:
Wouldn’t this mess up the results because the filter isn’t updated and the search not repeated?

Reply:
Sometimes (for example, AttachDialog) when the table is refreshed there is a call to refreshSearchList to repeat the search. Sometimes there isn't (//NOTE: DO NOT CALL mModuleList->refreshSearchList() IT WILL DEGRADE PERFORMANCE!).
If there is a filter, you don't need to call reloadData() at all since that will be invisible, if we keep the filter properly instead of clear it. Maybe StdSearchListView::reloadData() should be modified to do what refreshSearchList() does?
As for other methods, such as setRowCount , setCellContent and setRowIcon I have no idea why the filter must be cleared as the filter will be cleared or retained when reloadData() is called.